pkg/agent/secondarynetwork/podwatch/controller.go

```diff
@@ -686,6 +695,44 @@ func checkForPodSecondaryNetworkAttachment(pod *corev1.Pod) (string, bool) {
 	return netObj, netObjExists && netObj != ""
 }
 
+// When a Kubernetes Node reboots and the OVSDB file is not properly restored,
+// the primary and secondary OVS ports in OVSDB are lost. Pods may temporarily
+// appear as "unknown" and later get recreated with new container IDs.
+// The cleanup of secondary interfaces during agent restart depends on the InterfaceStores
+// being initialized with existing OVS ports. If the OVSDB file is missing, the
+// primary InterfaceStore is empty, preventing the secondary InterfaceStore from
+// initializing correctly. Consequently, secondary IPs assigned to old Pods (with
+// previous container IDs) are not deleted. To resolve this, all IPPool allocations
+// must be checked, and any IPs associated with non-existing container IDs should be
+// released. Otherwise, stale secondary IPs remain in the IPPool and cannot be reused.
+// We run it periodically to cover the case where a Pod is recreated with
+// the same name on another Node and has any secondary IP as well.
+func (pc *PodController) cleanUpStaleIPAddresses() {
+	pools, _ := pc.ipPoolLister.List(labels.Everything())
+	for _, ipPool := range pools {
+		for _, address := range ipPool.Status.IPAddresses {
+			if address.Owner.Pod == nil {
+				klog.V(2).InfoS("IPAM allocation found with no Pod owner")
```

Comment:
If this is really unexpected, shouldn't it be an Error or at least an Info message with default verbosity?

Also, we should include the IPPool reference in the logged values.